go/pkg/client/client.go

```diff
@@ -474,6 +474,12 @@ func getRPCCreds(ctx context.Context, credFile string, useApplicationDefault boo
 	return rpcCreds, CredsFileAuth, nil
 }
 
+// ByteStreamWriteOpts sets options required for resumable ByteStream write operations.
+type ByteStreamWriteOpts struct {
+	FinishWrite       bool
```

Comment:
What is the use-case for this option?

It looks like the new method sends all the given bytes to the server and tells the server to start writing at a specific offset. Since all the given bytes are sent, `req.FinishWrite` should always be sent at the end. No?

Reply:
The boolean flag is for the case where clients didn't send all bytes in the previous write requests and want to resume writing the rest of bytes. Yes it's always sent but I wanted to allow clients to set the flag based on their remaining data so that they can resume writing.

Reply:
That makes sense. Just noticed that the client associates the flag with the given resource name. Your implementation allows users to call the public method multiple times for the same resource name without causing errors, which makes sense with the new arbitrary offset option.